### DISCUSSION TITLE
Jakarta EE 10. Javax instead Jakarta

### DISCUSSION THREAD

**Discussion by Gio-Ayiman** [mapstruct/mapstruct#3338] [General]
I would like to know if any of you had a bug when generating mappers. Rather than importing jakarta.persistence.context it imports javax.persistence

**Comment by filiphr** [comment]
@Gio-Ayiman can you please be a bit more precise? What did you try? MapStruct does support Jakarta EE 10.

If you use the `jsr330` / `cdi` component model then MapStruct is going to use `javax` if those `javax` classes are on the classpath. Otherwise it is going to use `jakarta`.

If you have both `javax` and `jakarta` on the classpath then you should use `jakarta` / `jakarta-cdi` as component models.

**Reply by billerby** [reply]
I noticed that mapstruct seems to use the javax.annotation.Generated import regardless of component model used. I now have this in my compiler-plugin:


    <plugin>
          <artifactId>maven-compiler-plugin</artifactId>
          <version>${maven-compiler-plugin.version}</version>
          <configuration>
            <release>${jdk.version}</release>
            <annotationProcessorPaths>
              <path>
                <groupId>org.mapstruct</groupId>
                <artifactId>mapstruct-processor</artifactId>
                <version>${mapstruct.version}</version>
              </path>
            </annotationProcessorPaths>
            <showWarnings>true</showWarnings>
            <compilerArgs>
              <arg>
                -Amapstruct.suppressGeneratorTimestamp=true
              </arg>
              <arg>
                -Amapstruct.suppressGeneratorVersionInfoComment=true
              </arg>
              <arg>
                -Amapstruct.verbose=true
              </arg>
              <arg>
                -Amapstruct.defaultComponentModel=jakarta
              </arg>
            </compilerArgs>
          </configuration>
        </plugin>


And it only changes the inject import to jakarta:

        import jakarta.inject.Named;
        import jakarta.inject.Singleton;
        import javax.annotation.processing.Generated;
        
        @Generated(
            value = "org.mapstruct.ap.MappingProcessor"
        )
        @Singleton
        @Named
        public class TimeSlotMapperImpl implements TimeSlotMapper {


Is there something I am missing here? I have the jakarta.annotations on my classpath btw and using java21 (which has the javax.annotation on the classpath in its core).

**Reply by filiphr** [reply]
@billerby `javax.annotation.Generated` is a Java 8 JDK (https://docs.oracle.com/javase/8/docs/api/javax/annotation/Generated.html) annotation, and `javax.annotation.processing.Generated` is available from Java 9 (https://docs.oracle.com/javase/9/docs/api/javax/annotation/processing/Generated.html). They are both meant for the same thing and MapStruct is using one or the other depending on the source level of the project. When using Java 9+ then `javax.annotation.processing.Generated` will be used.

This has nothing to do with the component model
